src/cdk-experimental/menu/menu-bar.ts

```diff
@@ -207,6 +212,33 @@ export class CdkMenuBar extends CdkMenuGroup implements Menu, AfterContentInit,
     return this.orientation === 'horizontal';
   }
 
+  /**
+   * Subscribe to the menu trigger's open events in order to track the trigger which opened the menu
+   * and stop tracking it when the menu is closed closed.
+   */
+  private _subscribeToMenuOpen() {
+    this._allItems.changes
+      .pipe(
+        startWith(this._allItems),
+        mergeMap((list: QueryList<CdkMenuItem>) =>
+          list
+            .filter(item => item.hasMenu())
+            .map(item =>
+              item
+                .getMenuTrigger()!
+                .opened.pipe(mapTo(item), takeUntil(merge(this._allItems.changes, this._destroyed)))
```

Comment:
Here too, is it possible to unnest some of these nested operators? I've tended to have odd behavior with this sort of thing. (Should there be a persistent object-level observable merging this._allItems.changes and this._destroyed?)